modules/dxp/apps/osb/osb-faro/osb-faro-admin-web/src/main/java/com/liferay/osb/faro/admin/web/internal/portlet/action/DisconnectDataSourcesMVCActionCommand.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.osb.faro.admin.web.internal.portlet.action;
+
+import com.liferay.osb.faro.admin.web.internal.constants.FaroAdminPortletKeys;
+import com.liferay.osb.faro.model.FaroProject;
+import com.liferay.osb.faro.service.FaroProjectLocalService;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCActionCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCActionCommand;
+import com.liferay.portal.kernel.servlet.SessionErrors;
+import com.liferay.portal.kernel.util.Http;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Leslie Wong
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + FaroAdminPortletKeys.FARO_ADMIN,
+		"mvc.command.name=/faro_admin/disconnect_data_sources"
+	},
+	service = MVCActionCommand.class
+)
+public class DisconnectDataSourcesMVCActionCommand
+	extends BaseMVCActionCommand {
+
+	@Override
+	protected void doProcessAction(
+		ActionRequest actionRequest, ActionResponse actionResponse) {
+
+		long faroProjectId = ParamUtil.getLong(actionRequest, "faroProjectId");
+
+		try {
+			FaroProject faroProject = _faroProjectLocalService.getFaroProject(
+				faroProjectId);
+
+			Http.Options options = new Http.Options();
+
+			options.setLocation(
+				String.format(
+					"http://localhost:8080/o/faro/contacts/%s/data_source" +
```

Comment:
need to use portalURL

```
ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
	WebKeys.THEME_DISPLAY);
```

and then `String.format("%s/o/faro/contacts/%s/data_source/disconnect-all", themeDisplay.getPortalURL(), faroProject.getGroupId())`

Reply:
As the forwarded PR was merged, will send a follow-up PR to use the portal URL rather than hard-coding it.

Reply:
@rachaelkoestartyo Sent follow-up PR https://github.com/liferay-ac/liferay-portal/pull/1880